config/eslint.config.js

```diff
@@ -0,0 +1,54 @@
+import typescriptParser from '@typescript-eslint/parser'
+import typescriptEslint from '@typescript-eslint/eslint-plugin'
+
+export default [
+	{
+		files: ["**/*.ts"],
+		languageOptions: {
+			parser: typescriptParser,
+			parserOptions: {
+				project: './config/tsconfig.lint.json',
+				extraFileExtensions: ['.json'],
+				tsconfigRootDir: import.meta.dirname,
+			},
+			ecmaVersion: 2020,
+			sourceType: "module",
+		
+		},
+		plugins: {
+			'@typescript-eslint': typescriptEslint,
+		},
+		ignores: [
+			'node_modules/',
+			'dist/',
+			'dist.browser/',
+			'coverage/',
+			'prettier.config.cjs',
+			'typedoc.js',
+			'docs',
+			'.eslintrc.cjs',
+			'vitest.config.*.ts',
+			'archived-browser-client/',
+			'browser-client/',
+			'ui/',
+			'scripts/',
+			'proxy/',
+		],
+		rules: {
+			'@typescript-eslint/no-floating-promises': 'error',
+      '@typescript-eslint/no-misused-promises': 'error',
+      '@typescript-eslint/await-thenable': 'error',
+      '@typescript-eslint/no-unnecessary-type-assertion': 'error',
+      '@typescript-eslint/no-unsafe-argument': 'error',
+      '@typescript-eslint/no-unsafe-assignment': 'error',
+      '@typescript-eslint/no-unsafe-call': 'error',
+      '@typescript-eslint/no-unsafe-member-access': 'error',
+      '@typescript-eslint/no-unsafe-return': 'error',
```

Comment:
I'll be honest that I don't know the value of these but let's turn them all `off` for now.  They weren't triggering before and it's causing all sort of probably unnecessary lint issues.

Reply:
Seems since `recommended` is set to `true`, is either we explicitly turn off chosen recommended rules, but not `warn`